### PR TITLE
AOAI Structured Outputs Mode Enabled 

### PR DESCRIPTION
**Pull request by farzad528**
## Purpose

* The purpose of this PR is to create a complete `requirements.txt` file based on the latest package dependencies. This update is intended to avoid errors that were previously reported by customers. Additionally, the code has been modified to support OpenAI structured outputs, with a recommendation to use `gpt-4o` for these use cases. The README documentation has also been updated to provide guidance on using `gpt-4o-mini` by modifying the function to use non-structured outputs or regular JSON schema mode.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/farzad528/azure-search-power-skills
cd azure-search-power-skills
git checkout main
pip install -r requirements.txt
```

* Test the code

```
# Run the Azure Function locally
func start

# Test using the api-test.http file
Run the requests in the api-test.http file to validate the endpoints.
```

* Test the code

```
# Run the Azure Function locally
your-command-to-run-tests
```

## What to Check
Verify that the following are valid:
* The function runs without errors.
* Structured outputs are returned correctly when using `gpt-4o`.
* Documentation matches the updated functionality, including support for modifying the code to use `gpt-4o-mini` without structured outputs.

## Other Information

* Updated the dependencies to the latest versions based on customer feedback to enhance stability and compatibility.

### CODE REVIEW

File: AzureAIStudioCustomInferenceSkill/.funcignore
Comment (amitkalay) [inline AzureAIStudioCustomInferenceSkill/.funcignore:1]: Why did you delete these files? Is the code for using gpt-4o now exactly identical to the code we'd use for a custom language model like Phi, Mistral, etc.?

**Comment by farzad528** [comment]
Closing PR for now, will reopen with a new approach:
1) Will reinstate the AzureAIStudioLanguageModelSkill with guidance for customers to use this skill for modles such as Cohere, Phi-35, Mistral, and GPT-4o-mini, and GPT-4o. This repo will use the [Azure Inference API ](https://learn.microsoft.com/azure/ai-studio/reference/reference-model-inference-api?tabs=python).
2) Will refactor the AzureOpenAILanguageModelSkill to be exclusively for Azure OpenAI Service Models and use the [OpenAI Client SDK](https://learn.microsoft.com/azure/ai-services/openai/references/on-your-data?tabs=python)  supporting Structured Outputs since that's a powerful tool we should aim to leverage for customers using this capability. Also, will reinstate custom_prompts.json as I'm indifferent about this.
